### PR TITLE
Plone5 compat

### PR DESCRIPTION
**Pull request by cewing**
This PR adds plone 5 compatibility (and fixes a few small issues in the page templates around this).

### CODE REVIEW

File: Products/PloneKeywordManager/testing.py [hunk 1]
```diff
@@ -8,6 +8,15 @@
 from plone.testing import z2
 
 
+try:
```
Comment (hvelarde) [inline Products/PloneKeywordManager/testing.py]: use the API: http://docs.plone.org/develop/plone.api/docs/env.html#plone-version

File: Products/PloneKeywordManager/testing.py [hunk 2]
```diff
@@ -27,6 +39,9 @@ def setUpPloneSite(self, portal):
         self.applyProfile(portal, 'Products.PloneKeywordManager:default')
         self.applyProfile(portal, 'plone.app.dexterity:default')
 
+        if PLONE_5:
```
Comment (hvelarde) [inline Products/PloneKeywordManager/testing.py]: Plone 5 should test against plone.app.contenttypes

File: setup.py [hunk 3]
```diff
@@ -47,6 +48,8 @@
         'test': [
           'plone.app.testing',
           'plone.app.dexterity',
+          'Products.ATContentTypes',
+          'unittest2',
```
Comment (hvelarde) [inline setup.py]: unittest2 is for Python 2.6 compatibility; get rid of it.

File: Products/PloneKeywordManager/testing.py (hunk 1 above)
Comment (cewing) [inline Products/PloneKeywordManager/testing.py]: Oh thank you!  I knew there had to be a saner way to do this!

File: Products/PloneKeywordManager/testing.py (hunk 2 above)
Comment (cewing) [inline Products/PloneKeywordManager/testing.py]: I believe it does.  I think the point here is to ensure that AT support is available under Plone 5, where it no longer is installed by default.  Without including this, the tests fail, but dexterity tests pass under both Plone 4 and Plone 5 already.

File: setup.py (hunk 3 above)
Comment (cewing) [inline setup.py]: We still have envs in our travis matrix for Plone 4.2 and 4.1 with Python 2.6 support.  I'm not okay with dropping that unless we are saying explicitly that we no longer support 4.2.

File: Products/PloneKeywordManager/testing.py (hunk 2 above)
Comment (hvelarde) [inline Products/PloneKeywordManager/testing.py]: IMO Plone 4.x should test against Archetypes and Plone 5.x should test against plone.app.contenttypes only; that would keep the test fixture clean and sane.

File: setup.py (hunk 3 above)
Comment (cewing) [inline setup.py]: And to be clear @hvelarde, I agree that we can drop that support, but we need to make it explicit

File: Products/PloneKeywordManager/testing.py (hunk 2 above)
Comment (cewing) [inline Products/PloneKeywordManager/testing.py]: That will take a larger refactoring of the test code that I was ready to take on at this point @hvelarde.  Do you think this is a requirement for merging?

File: setup.py (hunk 3 above)
Comment (hvelarde) [inline setup.py]: Plone 4.2 runs under Python 2.7; Python 2.6 should be deprecated on tests and people wanting to use it in production must be aware of that.

File: Products/PloneKeywordManager/testing.py (hunk 2 above)
Comment (hvelarde) [inline Products/PloneKeywordManager/testing.py]: no; please open a new issue so we can work on that later :-)

**Comment by cewing** [comment]
@hvelarde, I've fixed using `plone.api` for version detection both in test setup and also now in showing proper search icons for keywords in Plone 4 and in Plone 5.

I also lifted the accessibility work performed by @daniele-andreotti on PR #13, so if we merge this PR we can close that one as the work there is included here too.

Let's merge this and do a release so we've got one more Plone 5 compatible add-on out there.

**Comment by hvelarde** [comment]
@cewing do you mind to squash the whole thing into one commit?

**Comment by jensens** [comment]
Dont let code get old in PRs. This one is a huge step forward. I merge.

**Comment by cewing** [comment]
shoot.  Somehow I missed your request above, @hvelarde.  Thanks very much to @jensens for merging anyway.  I appreciate it.

**Pull request merged**